src/core/core.controller.js

```diff
@@ -1026,23 +1026,24 @@ class Chart {
 	/**
 	 * Handle an event
 	 * @param {IEvent} e the event to handle
+	 * @param {boolean} [replay] - true if the event was replayed by `update`
 	 * @return {boolean} true if the chart needs to re-render
 	 * @private
 	 */
-	_handleEvent(e) {
+	_handleEvent(e, replay) {
 		const me = this;
-		const options = me.options || {};
+		const options = me.options;
 		const hoverOptions = options.hover;
+		// If the event is replayed from `update`, we should evaluate with the final positions.
```

Comment:
Could you expand this comment a little? There's a lot going on here, so I just want to make sure I understand correctly.

I think the following is perhaps true:
* It's referring the final position after this `update`
* The event will have mouse coordinates from the last event. This should basically be where the mouse is currently

Why do we handle the event at the final position? Once the animation starts it will move it back to the initial position and then animate it back towards the final position?